### PR TITLE
Refactor `FlowControlDataQueue` to improve performances

### PR DESCRIPTION
**Pull request by bdraco**
The `DataQueue` base class tracks `size` but it never uses it. Move all the `size` tracking into the `FlowControlDataQueue`

### CODE REVIEW

**Comment by bdraco** [comment]
why is that super call so expensive.....

File: aiohttp/streams.py [hunk 1]
```diff
@@ -698,8 +696,23 @@ def feed_data(self, data: _SizedT) -> None:
             self._protocol.pause_reading()
 
     async def read(self) -> _SizedT:
-        try:
-            return await super().read()
```
Comment (bdraco) [inline aiohttp/streams.py:702]: This super call is more expensive than I would expect.

... writing this out twice is not good though....

need to come up with something better

File: aiohttp/streams.py
Comment (Dreamsorcerer) [inline aiohttp/streams.py:1]: As DataQueue doesn't have _size anymore, the Generic should not require Sized anymore. Also, probably want to have WSMessage queues move to that, as they don't need the _size calculations.

Comment (bdraco) [reply aiohttp/streams.py:1]: `_SizedT` is in this case `WSMessage` .... I think could be changed to `_T`

`FlowControlDataQueue` is only used by the WebSocket?  Does the WebSocket not need to pause the protocol when the limit is hit?

Comment (bdraco) [reply aiohttp/streams.py:1]: https://github.com/aio-libs/aiohttp/pull/9677

Comment (Dreamsorcerer) [reply aiohttp/streams.py:1]: Oh yeah. That logic can simply be removed then. I thought I remembered it being used in the parser or something with bytes, where it tracked the size of the data stream..

**Pull request merged**